server/svix-server/src/main.rs

```diff
@@ -53,35 +61,7 @@ enum JwtCommands {
     Generate,
 }
 
-#[tokio::main]
-async fn main() {
-    dotenv().ok();
-
-    let args = Args::parse();
-    let cfg = cfg::load().unwrap();
-
-    if cfg!(debug_assertions) && std::env::var_os("RUST_LOG").is_none() {
-        std::env::set_var(
-            "RUST_LOG",
-            format!(
-                "{crate}={level},tower_http={level}",
-                crate = CRATE_NAME,
-                level = cfg.log_level.to_string()
-            ),
-        );
-    }
-
-    tracing_subscriber::fmt::init();
-
-    if let Some(Commands::Jwt {
-        command: JwtCommands::Generate,
-    }) = &args.command
-    {
-        let token = generate_token(&cfg.jwt_secret).unwrap();
-        println!("Token (Bearer): {}", token);
-        exit(0);
-    }
-
+pub(crate) async fn run(cfg: Configuration, listener: Option<TcpListener>) {
```

Comment:
No particular objection, but why do we need to potentially pass listener? Why can't we just pass the custom URL in the config in the tests?

Reply:
I wanted to pass a `TcpListener` so I could bind to port `0` and get the one that was actually assigned via the `local_addr` method. I figured this was easier than requiring a port block be free for all the test instances.

Reply:
Yeah, cool, you're 100% right. :)